src/modules/vision_target_estimator/Orientation/KF_orientation_moving.h

```diff
@@ -0,0 +1,114 @@
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file KF_orientation_moving.h
+ * @brief Filter to estimate the orientation of moving targets. State: [theta, theta_dot]
+ *
+ * @author Jonas Perolini <jonspero@me.com>
+ *
+ */
+
+#include <matrix/math.hpp>
+#include <mathlib/mathlib.h>
+#include <matrix/Matrix.hpp>
+#include <matrix/Vector.hpp>
+
+#include "base_KF_orientation.h"
+
+#pragma once
+
+namespace vision_target_estimator
+{
+class KF_orientation_moving : public Base_KF_orientation
+{
+public:
+	/**
+	 * Default constructor, state not initialized
+	 */
+	KF_orientation_moving() {};
+
+	/**
+	 * Default desctructor
+	 */
+	virtual ~KF_orientation_moving() {};
+
+	//Prediction step:
+	void predictState(float dt) override;
+	void predictCov(float dt) override;
+
+	// Backwards state prediciton
+	void syncState(float dt) override;
+
+	void setH(matrix::Vector<float, 2> h_meas) override;
+
+	virtual float computeInnovCov(float measUnc) override;
+	virtual float computeInnov(float meas) override;
+
+	bool update() override;
+
+	void setNISthreshold(float nis_threshold) override { _nis_threshold = nis_threshold; };
+
+	// Init: x_0
+	void setPosition(float pos) override { _state(0, 0) = pos; };
+	void setVelocity(float vel) override { _state(1, 0) = vel; };
+
+	// Init: P_0
+	void setStatePosVar(float pos_unc) override { _covariance(0, 0) = pos_unc; };
+	void setStateVelVar(float vel_unc) override { _covariance(1, 1) = vel_unc; };
+
+	// Retreive output of filter
+	float getPosition() override { return _state(0, 0); };
+	float getVelocity() override { return _state(1, 0); };
+
+	float getPosVar() override { return _covariance(0, 0); };
+	float getVelVar() override { return _covariance(1, 1); };
+
+	float getTestRatio() override {if (fabsf(_innov_cov) < 1e-6f) {return -1.f;} else {return _innov / _innov_cov * _innov;} };
+
+private:
+	matrix::Matrix<float, 2, 1> _state; // state
+
+	matrix::Matrix<float, 2, 1> _sync_state; // state
+
+	matrix::Matrix<float, 1, 2> _meas_matrix; // row of measurement matrix
```

Comment:
Those matrices with 1 column could just be vectors, right?

Reply:
Done for `_state` and `_sync_state` in https://github.com/PX4/PX4-Autopilot/pull/22008/commits/cc0e00bf4936b15bc79108caac84007babc6aa67. Not done for `_meas_matrix` because it is used as a matrix in Symforce so it requires more work. Please let me know if you think it's necessary.

Reply:
Also done for the meas matrix here https://github.com/PX4/PX4-Autopilot/commit/f32d10096eede140a8e44e07245113ab9b366d11